src/org/labkey/targetedms/TargetedMSController.java

```diff
@@ -1508,12 +1512,26 @@ else if ("pdf".equalsIgnoreCase(form.getFormat()))
     }
 
     @NotNull
-    private String renderSVG(AbstractChartForm form, JFreeChart chart)
+    private String renderSVG(AbstractChartForm form, JFreeChart chart) throws SVGGraphics2DIOException
     {
-        SVGGraphics2D g2 = new SVGGraphics2D(form.getChartWidth(), form.getChartHeight());
-        Rectangle r = new Rectangle(0, 0, form.getChartWidth(), form.getChartHeight());
-        chart.draw(g2, r);
-        return g2.getSVGElement();
+        DOMImplementation domImpl = GenericDOMImplementation.getDOMImplementation();
+
+        // Create an instance of org.w3c.dom.Document.
+        String svgNS = "http://www.w3.org/2000/svg";
+        Document document = domImpl.createDocument(svgNS, "svg", null);
+
+        // Create an instance of the SVG Generator from Batik
+        SVGGraphics2D svgGenerator = new SVGGraphics2D(document);
+        Dimension dim = new Dimension(form.getChartWidth(), form.getChartHeight());
+        svgGenerator.setSVGCanvasSize(dim);
+        Rectangle r = new Rectangle(dim);
+
+        chart.draw(svgGenerator, r);
+
+        // Render SVG to a string
+        StringWriter out = new StringWriter();
```

Comment:
should it be inside try-with-resources? I guess not as it is not actually holding any underlying resources ?

Reply:
Yeah, since it's a StringWriter I didn't bother.